processor/cumulativetodeltaprocessor/internal/tracking/tracker.go

```diff
@@ -161,12 +161,25 @@ func (t *MetricTracker) Convert(in MetricPoint) (out DeltaValue, valid bool) {
 
 		delta := value.Clone()
 
-		// Calculate deltas unless histogram count was reset
+		// Calculate deltas unless the histogram count was reset, in which case
+		// take the current data point as the delta.
 		if valid && delta.Count >= prevValue.Count {
-			delta.Count -= prevValue.Count
-			delta.Sum -= prevValue.Sum
+			// It's possible the overall count remained the same after a reset,
+			// so verify each bucket. Histogram bucket values cannot decrease,
+			// so a lower value indicates a reset.
+			reset := false
 			for index, prevBucket := range prevValue.Buckets {
-				delta.Buckets[index] -= prevBucket
+				if prevBucket > delta.Buckets[index] {
+					reset = true
```

Comment:
I think at this point we can break the loop, we don't need to check all buckets, we just need to check if one of them is invalid.